_studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw.cpp

```diff
@@ -1031,8 +1031,8 @@ mfxStatus MFXVideoENCODEMJPEG_HW::CheckEncodeFrameParam(
             MFX_CHECK(surface->Data.Y != 0 || isExternalFrameAllocator, MFX_ERR_UNDEFINED_BEHAVIOR);
         }
 
-        if (surface->Info.Width != m_vParam.mfx.FrameInfo.Width || surface->Info.Height != m_vParam.mfx.FrameInfo.Height)
-            sts = MFX_WRN_INCOMPATIBLE_VIDEO_PARAM;
+        MFX_CHECK(surface->Info.Width >= m_vParam.mfx.FrameInfo.Width, MFX_ERR_INVALID_VIDEO_PARAM);
```

Comment:
Warning changed to error. Looks like a typo.

Reply:
@stasefremov  I think it would be better to align the behavior with other CODECs, E.g. for H264,  MFX_ERR_INVALID_VIDEO_PARAM is returned when the condition is false (https://github.com/Intel-Media-SDK/MediaSDK/blob/f3b12a7ce55568be212a42a36257de2abc8150f2/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp#L3021-L3022)

Reply:
I agree with Haihao

Reply:
It would be better to return MFX_ERR_INVALID_VIDEO_PARAM for jpeg too when the condition is false. If doing so, we may use the same error handling path in user application.  How do you think? @dmitryermilov @stasefremov

Reply:
In addition, I don't think it is suitable to return MFX_WRN_INCOMPATIBLE_VIDEO_PARAM when the surface size is smaller than the video frame size.